internal/provider/resource/integration_aws_secrets_manager.go

```diff
@@ -0,0 +1,538 @@
+
+// Metadata returns the resource type name.
+func (r *IntegrationAWSSecretsManagerResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_integration_aws_secrets_manager"
+}
+
+// Schema defines the schema for the resource.
+func (r *IntegrationAWSSecretsManagerResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Create AWS Secrets Manager integration & save to Infisical. Only Machine Identity authentication is supported for this data source",
+		Attributes: map[string]schema.Attribute{
+			"options": schema.SingleNestedAttribute{
+				Description: "Integration options",
+				Optional:    true,
+				Attributes: map[string]schema.Attribute{
+					"secret_prefix": schema.StringAttribute{
+						Optional:    true,
+						Description: "The prefix to add to the secret name in AWS Secrets Manager.",
+					},
+
+					"aws_tags": schema.ListNestedAttribute{
+						Description:   "Tags to attach to the AWS Secrets Manager secrets.",
+						Optional:      true,
+						PlanModifiers: []planmodifier.List{listplanmodifier.RequiresReplace()},
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"key": schema.StringAttribute{
+									Description:   "The key of the tag.",
+									Optional:      true,
+									PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+								},
+								"value": schema.StringAttribute{
+									Description:   "The value of the tag.",
+									Optional:      true,
+									PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"integration_auth_id": schema.StringAttribute{
+				Computed:      true,
+				Description:   "The ID of the integration auth, used internally by Infisical.",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+
+			"integration_id": schema.StringAttribute{
+				Computed:      true,
+				Description:   "The ID of the integration, used internally by Infisical.",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+
+			"aws_region": schema.StringAttribute{
+				Required:      true,
+				Description:   "The AWS region to sync secrets to.",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+
+			"access_key_id": schema.StringAttribute{
+				Sensitive:     true,
+				Required:      true,
+				Description:   "The AWS access key ID.",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+
+			"secret_access_key": schema.StringAttribute{
+				Sensitive:     true,
+				Required:      true,
+				Description:   "The AWS secret access key.",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
```

Comment:
why does updating these require a complete replace?

Reply:
Because the project ID cannot be updated, since the user shouldn't be moving integrations out of the project it was initially created in. If they want to do that, a new integration should take it's place

Reply:
Because it would change the auth itself, which we currently don't support updating